In [ ]:
import pandas as pd
from script_completion import print_results

In [2]:
data = pd.read_csv("data.csv")

In [3]:
data.head()

preffix  \
0  import pandas as pd\nimport numpy as np\nimpor...   
1  import pandas as pd\nimport numpy as np\nimpor...   
2  import pandas as pd\nimport numpy as np\nimpor...   
3  import pandas as pd\nimport numpy as np\nimpor...   
4  import pandas as pd\nimport numpy as np\nimpor...   

                                              suffix  \
0      def forward(self, x):\n        x = x.flatt...   
1              x = self.conv1_drop(x)\n        x ...   
2  model_conv = ConvNetwork(training_loader, vali...   
3          x = F.relu(self.fc2(x))\n        x = s...   
4  \n\nmodel_dense = DenseNetwork(training_loader...   

                           label                                  result  \
0                # dopredny chod                   def forward(self, x):   
1          if self.dropout_prob:          x = F.relu(self.conv1_drop(x))   
2                            NaN                                     NaN   
3               x = self.drop(x)                        x = self.drop(x)   
4                                                                    NaN   

         CHRF exact_match correct_syntax  correct_code    rougeL  
0    5.868545       False          False         False  0.000000  
1   23.582904       False           True         False  0.181818  
2    0.000000         NaN            NaN          True  0.000000  
3  100.000000        True           True          True  1.000000  
4    0.000000       False           True          True  0.000000

In [6]:
def show_data(data):
    print(f"\tPrefix:\n{data['preffix']}\n\n\tFill here\n\tActual code:\n{data['label']}\n\tGenerated code:\n{data['result']}\n\nSuffix:\n{data['suffix']}")
    

In [7]:
show_data(data.iloc[0])

	Prefix:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from torch import nn
import torch.nn.functional as F
import torch.utils.data
import matplotlib.pyplot as plt
import Network

data = pd.read_csv("train.csv")

label = data['label']
img = data.drop('label', axis=1)
img_tensor = np.array(img).reshape(-1, 1, 32, 32)

dataset = torch.utils.data.TensorDataset(torch.tensor(img_tensor, dtype = torch.float).to("cuda"), torch.tensor(label, dtype = torch.long).to("cuda"))
generator = torch.Generator().manual_seed(17)
train_data, val_data, test_data = torch.utils.data.random_split(dataset, [0.7, 0.15, 0.15], generator)

training_loader = torch.utils.data.DataLoader(train_data, batch_size=32, shuffle=True)
validation_loader = torch.utils.data.DataLoader(val_data, batch_size=128, shuffle=False)
testing_loader = torch.utils.data.DataLoader(test_data, batch_size=128, shuffle=False)
        

class DenseNetwork(Network):
    def __init__(self, training_loader, validation_lo

wrong code - duplicates a line that is used afterwards

In [8]:
data.loc[0, "correct_code"] = False

In [9]:
show_data(data.iloc[1])

	Prefix:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from torch import nn
import torch.nn.functional as F
import torch.utils.data
import matplotlib.pyplot as plt
import Network

data = pd.read_csv("train.csv")

label = data['label']
img = data.drop('label', axis=1)
img_tensor = np.array(img).reshape(-1, 1, 32, 32)

dataset = torch.utils.data.TensorDataset(torch.tensor(img_tensor, dtype = torch.float).to("cuda"), torch.tensor(label, dtype = torch.long).to("cuda"))
generator = torch.Generator().manual_seed(17)
train_data, val_data, test_data = torch.utils.data.random_split(dataset, [0.7, 0.15, 0.15], generator)

training_loader = torch.utils.data.DataLoader(train_data, batch_size=32, shuffle=True)
validation_loader = torch.utils.data.DataLoader(val_data, batch_size=128, shuffle=False)
testing_loader = torch.utils.data.DataLoader(test_data, batch_size=128, shuffle=False)
        

class DenseNetwork(Network):
    def __init__(self, training_loader, validation_lo

wrong code - applies relu, where it should not have been

In [10]:
data.loc[1, "correct_code"] = False

In [11]:
show_data(data.iloc[2])

	Prefix:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from torch import nn
import torch.nn.functional as F
import torch.utils.data
import matplotlib.pyplot as plt
import Network

data = pd.read_csv("train.csv")

label = data['label']
img = data.drop('label', axis=1)
img_tensor = np.array(img).reshape(-1, 1, 32, 32)

dataset = torch.utils.data.TensorDataset(torch.tensor(img_tensor, dtype = torch.float).to("cuda"), torch.tensor(label, dtype = torch.long).to("cuda"))
generator = torch.Generator().manual_seed(17)
train_data, val_data, test_data = torch.utils.data.random_split(dataset, [0.7, 0.15, 0.15], generator)

training_loader = torch.utils.data.DataLoader(train_data, batch_size=32, shuffle=True)
validation_loader = torch.utils.data.DataLoader(val_data, batch_size=128, shuffle=False)
testing_loader = torch.utils.data.DataLoader(test_data, batch_size=128, shuffle=False)
        

class DenseNetwork(Network):
    def __init__(self, training_loader, validation_lo

correct code - empty line

In [12]:
data.loc[2, "correct_code"] = True

In [13]:
show_data(data.iloc[3])

	Prefix:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from torch import nn
import torch.nn.functional as F
import torch.utils.data
import matplotlib.pyplot as plt
import Network

data = pd.read_csv("train.csv")

label = data['label']
img = data.drop('label', axis=1)
img_tensor = np.array(img).reshape(-1, 1, 32, 32)

dataset = torch.utils.data.TensorDataset(torch.tensor(img_tensor, dtype = torch.float).to("cuda"), torch.tensor(label, dtype = torch.long).to("cuda"))
generator = torch.Generator().manual_seed(17)
train_data, val_data, test_data = torch.utils.data.random_split(dataset, [0.7, 0.15, 0.15], generator)

training_loader = torch.utils.data.DataLoader(train_data, batch_size=32, shuffle=True)
validation_loader = torch.utils.data.DataLoader(val_data, batch_size=128, shuffle=False)
testing_loader = torch.utils.data.DataLoader(test_data, batch_size=128, shuffle=False)
        

class DenseNetwork(Network):
    def __init__(self, training_loader, validation_lo

correct code - exact match

In [14]:
data.loc[3, "correct_code"] = True

In [15]:
show_data(data.iloc[4])

	Prefix:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from torch import nn
import torch.nn.functional as F
import torch.utils.data
import matplotlib.pyplot as plt
import Network

data = pd.read_csv("train.csv")

label = data['label']
img = data.drop('label', axis=1)
img_tensor = np.array(img).reshape(-1, 1, 32, 32)

dataset = torch.utils.data.TensorDataset(torch.tensor(img_tensor, dtype = torch.float).to("cuda"), torch.tensor(label, dtype = torch.long).to("cuda"))
generator = torch.Generator().manual_seed(17)
train_data, val_data, test_data = torch.utils.data.random_split(dataset, [0.7, 0.15, 0.15], generator)

training_loader = torch.utils.data.DataLoader(train_data, batch_size=32, shuffle=True)
validation_loader = torch.utils.data.DataLoader(val_data, batch_size=128, shuffle=False)
testing_loader = torch.utils.data.DataLoader(test_data, batch_size=128, shuffle=False)
        

class DenseNetwork(Network):
    def __init__(self, training_loader, validation_lo

correct code - empty line

In [16]:
data.loc[4, "correct_code"] = True

In [17]:
show_data(data.iloc[5])

	Prefix:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from torch import nn
import torch.nn.functional as F
import torch.utils.data
import matplotlib.pyplot as plt
import Network

data = pd.read_csv("train.csv")

label = data['label']
img = data.drop('label', axis=1)
img_tensor = np.array(img).reshape(-1, 1, 32, 32)

dataset = torch.utils.data.TensorDataset(torch.tensor(img_tensor, dtype = torch.float).to("cuda"), torch.tensor(label, dtype = torch.long).to("cuda"))
generator = torch.Generator().manual_seed(17)
train_data, val_data, test_data = torch.utils.data.random_split(dataset, [0.7, 0.15, 0.15], generator)

training_loader = torch.utils.data.DataLoader(train_data, batch_size=32, shuffle=True)
validation_loader = torch.utils.data.DataLoader(val_data, batch_size=128, shuffle=False)
testing_loader = torch.utils.data.DataLoader(test_data, batch_size=128, shuffle=False)
        

class DenseNetwork(Network):
    def __init__(self, training_loader, validation_lo

correct code - empty line

In [18]:
data.loc[5, "correct_code"] = True

In [19]:
show_data(data.iloc[6])

	Prefix:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

	Fill here
	Actual code:
from torch import nn
	Generated code:
import torch

Suffix:
import torch.nn.functional as F
import torch.utils.data
import matplotlib.pyplot as plt
import Network

data = pd.read_csv("train.csv")

label = data['label']
img = data.drop('label', axis=1)
img_tensor = np.array(img).reshape(-1, 1, 32, 32)

dataset = torch.utils.data.TensorDataset(torch.tensor(img_tensor, dtype = torch.float).to("cuda"), torch.tensor(label, dtype = torch.long).to("cuda"))
generator = torch.Generator().manual_seed(17)
train_data, val_data, test_data = torch.utils.data.random_split(dataset, [0.7, 0.15, 0.15], generator)

training_loader = torch.utils.data.DataLoader(train_data, batch_size=32, shuffle=True)
validation_loader = torch.utils.data.DataLoader(val_data, batch_size=128, shuffle=False)
testing_loader = torch.utils.data.DataLoader(test_data, batch_size=128, shuffle=False)
        

class DenseNetwor

almost correct - importing correct library, but the code still will not work

In [20]:
data.loc[6, "correct_code"] = False

In [21]:
show_data(data.iloc[7])

	Prefix:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from torch import nn
import torch.nn.functional as F
import torch.utils.data
import matplotlib.pyplot as plt
import Network

data = pd.read_csv("train.csv")

label = data['label']
img = data.drop('label', axis=1)
img_tensor = np.array(img).reshape(-1, 1, 32, 32)

dataset = torch.utils.data.TensorDataset(torch.tensor(img_tensor, dtype = torch.float).to("cuda"), torch.tensor(label, dtype = torch.long).to("cuda"))
generator = torch.Generator().manual_seed(17)
train_data, val_data, test_data = torch.utils.data.random_split(dataset, [0.7, 0.15, 0.15], generator)

training_loader = torch.utils.data.DataLoader(train_data, batch_size=32, shuffle=True)
validation_loader = torch.utils.data.DataLoader(val_data, batch_size=128, shuffle=False)
testing_loader = torch.utils.data.DataLoader(test_data, batch_size=128, shuffle=False)
        

class DenseNetwork(Network):
    def __init__(self, training_loader, validation_lo

correct code - empty line

In [22]:
data.loc[7, "correct_code"] = True

In [23]:
show_data(data.iloc[8])

	Prefix:
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error, mean_squared_error
import math
import pandas as pd
from sklearn.linear_model import Ridge

def get_model(data):
    y = data['days'].dt.total_seconds() / 60

	Fill here
	Actual code:
    X = data.drop(columns='days')
	Generated code:
    X = data['days'].dt.total_seconds() / 60

Suffix:
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

    model = Ridge().fit(X_train, y_train)

    y_pred = model.predict(X_test)
    acc = mean_absolute_error(y_pred, y_test)
    coef = model.coef_
    print("Acc:", acc)
    print("Coefs:", coef)

if __name__ == "__main__":
    interested = pd.read_csv("data.csv")
    get_model(interested)


wrong code - selecting opposite columns from the dataset

In [24]:
data.loc[8, "correct_code"] = False

In [25]:
show_data(data.iloc[9])

	Prefix:
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error, mean_squared_error
import math
import pandas as pd
from sklearn.linear_model import Ridge

	Fill here
	Actual code:
nan
	Generated code:
nan

Suffix:
def get_model(data):
    y = data['days'].dt.total_seconds() / 60
    X = data.drop(columns='days')
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

    model = Ridge().fit(X_train, y_train)

    y_pred = model.predict(X_test)
    acc = mean_absolute_error(y_pred, y_test)
    coef = model.coef_
    print("Acc:", acc)
    print("Coefs:", coef)

if __name__ == "__main__":
    interested = pd.read_csv("data.csv")
    get_model(interested)


correct code - empty line

In [26]:
data.loc[9, "correct_code"] = True

In [27]:
show_data(data.iloc[10])

	Prefix:
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error, mean_squared_error
import math
import pandas as pd
from sklearn.linear_model import Ridge

def get_model(data):
    y = data['days'].dt.total_seconds() / 60

	Fill here
	Actual code:
    X = data.drop(columns='days')
	Generated code:
    X = data['days'].dt.total_seconds() / 60

Suffix:
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

    model = Ridge().fit(X_train, y_train)

    y_pred = model.predict(X_test)
    acc = mean_absolute_error(y_pred, y_test)
    coef = model.coef_
    print("Acc:", acc)
    print("Coefs:", coef)

if __name__ == "__main__":
    interested = pd.read_csv("data.csv")
    get_model(interested)


wrong code - selecting opposite columns

In [28]:
data.loc[10, "correct_code"] = False

In [29]:
show_data(data.iloc[11])

	Prefix:
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error, mean_squared_error

	Fill here
	Actual code:
import math
	Generated code:
from sklearn.metrics <|endoftext|>

Suffix:
import pandas as pd
from sklearn.linear_model import Ridge

def get_model(data):
    y = data['days'].dt.total_seconds() / 60
    X = data.drop(columns='days')
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

    model = Ridge().fit(X_train, y_train)

    y_pred = model.predict(X_test)
    acc = mean_absolute_error(y_pred, y_test)
    coef = model.coef_
    print("Acc:", acc)
    print("Coefs:", coef)

if __name__ == "__main__":
    interested = pd.read_csv("data.csv")
    get_model(interested)


wrong code - importing wrong library

In [30]:
data.loc[11, "correct_code"] = False

In [31]:
show_data(data.iloc[12])

	Prefix:
from sklearn.linear_model import LinearRegression

	Fill here
	Actual code:
from sklearn.model_selection import train_test_split
	Generated code:
nan

Suffix:
from sklearn.metrics import mean_absolute_error, mean_squared_error
import math
import pandas as pd
from sklearn.linear_model import Ridge

def get_model(data):
    y = data['days'].dt.total_seconds() / 60
    X = data.drop(columns='days')
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

    model = Ridge().fit(X_train, y_train)

    y_pred = model.predict(X_test)
    acc = mean_absolute_error(y_pred, y_test)
    coef = model.coef_
    print("Acc:", acc)
    print("Coefs:", coef)

if __name__ == "__main__":
    interested = pd.read_csv("data.csv")
    get_model(interested)


wrong code - wrong library selected

In [32]:
data.loc[12, "correct_code"] = False

In [33]:
show_data(data.iloc[13])

	Prefix:
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error, mean_squared_error
import math
import pandas as pd
from sklearn.linear_model import Ridge


	Fill here
	Actual code:
def get_model(data):
	Generated code:
nan

Suffix:
    y = data['days'].dt.total_seconds() / 60
    X = data.drop(columns='days')
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

    model = Ridge().fit(X_train, y_train)

    y_pred = model.predict(X_test)
    acc = mean_absolute_error(y_pred, y_test)
    coef = model.coef_
    print("Acc:", acc)
    print("Coefs:", coef)

if __name__ == "__main__":
    interested = pd.read_csv("data.csv")
    get_model(interested)


wrong code - no start of a function definition generated

In [34]:
data.loc[13, "correct_code"] = False

In [35]:
show_data(data.iloc[14])

	Prefix:
from sklearn.linear_model import LinearRegression

	Fill here
	Actual code:
from sklearn.model_selection import train_test_split
	Generated code:
nan

Suffix:
from sklearn.metrics import mean_absolute_error, mean_squared_error
import math
import pandas as pd
from sklearn.linear_model import Ridge

def get_model(data):
    y = data['days'].dt.total_seconds() / 60
    X = data.drop(columns='days')
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

    model = Ridge().fit(X_train, y_train)

    y_pred = model.predict(X_test)
    acc = mean_absolute_error(y_pred, y_test)
    coef = model.coef_
    print("Acc:", acc)
    print("Coefs:", coef)

if __name__ == "__main__":
    interested = pd.read_csv("data.csv")
    get_model(interested)


wrong code - library not imported

In [36]:
data.loc[14, "correct_code"] = False

In [37]:
show_data(data.iloc[15])

	Prefix:
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split

	Fill here
	Actual code:
from sklearn.metrics import mean_absolute_error, mean_squared_error
	Generated code:
from sklearn.metrics <|endoftext|>

Suffix:
import math
import pandas as pd
from sklearn.linear_model import Ridge

def get_model(data):
    y = data['days'].dt.total_seconds() / 60
    X = data.drop(columns='days')
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

    model = Ridge().fit(X_train, y_train)

    y_pred = model.predict(X_test)
    acc = mean_absolute_error(y_pred, y_test)
    coef = model.coef_
    print("Acc:", acc)
    print("Coefs:", coef)

if __name__ == "__main__":
    interested = pd.read_csv("data.csv")
    get_model(interested)


wrong code - wrong import of a library

In [38]:
data.loc[15, "correct_code"] = False

In [39]:
show_data(data.iloc[16])

	Prefix:
'''Filter'''
import numpy as np

def get_center(kernel):
    '''Function that gets the center of the kernel'''
    if kernel.shape[0] % 2 == 1 and kernel.shape[1] % 2 == 1:
        return [kernel.shape[0]//2, kernel.shape[1]//2]
    return [0, 0]


def get_padding(center, kernel):
    '''Gets number of zeros to be padded in each direction'''
    if center == [0,0]:
        return ((1,1),(1,1))
    return (center[0], kernel.shape[0]-1-center[0]), (center[1], kernel.shape[1]-1-center[1])


def convolve(array, kernel):
    '''Applies convolution to a array'''
    center = get_center(kernel)
    padding = get_padding(center, kernel)
    pad_image = np.pad(array, padding)
    sha = (array.shape[0], array.shape[1], kernel.shape[0], kernel.shape[1])
    stride = np.lib.stride_tricks.as_strided(pad_image, shape=sha, strides=pad_image.strides*2)
    return np.sum(stride * kernel, axis=(2, 3))


def apply_filter(image: np.array, kernel: np.array) -> np.array:
    '''Applies filter on a 

correct code - convolving red channel

In [40]:
data.loc[16, "correct_code"] = True

In [41]:
show_data(data.iloc[17])

	Prefix:
'''Filter'''
import numpy as np


	Fill here
	Actual code:
def get_center(kernel):
	Generated code:
nan

Suffix:
    '''Function that gets the center of the kernel'''
    if kernel.shape[0] % 2 == 1 and kernel.shape[1] % 2 == 1:
        return [kernel.shape[0]//2, kernel.shape[1]//2]
    return [0, 0]


def get_padding(center, kernel):
    '''Gets number of zeros to be padded in each direction'''
    if center == [0,0]:
        return ((1,1),(1,1))
    return (center[0], kernel.shape[0]-1-center[0]), (center[1], kernel.shape[1]-1-center[1])


def convolve(array, kernel):
    '''Applies convolution to a array'''
    center = get_center(kernel)
    padding = get_padding(center, kernel)
    pad_image = np.pad(array, padding)
    sha = (array.shape[0], array.shape[1], kernel.shape[0], kernel.shape[1])
    stride = np.lib.stride_tricks.as_strided(pad_image, shape=sha, strides=pad_image.strides*2)
    return np.sum(stride * kernel, axis=(2, 3))


def apply_filter(image: np.array, ke

wrong code - start of a function definition not generated

In [42]:
data.loc[17, "correct_code"] = False

In [43]:
show_data(data.iloc[18])

	Prefix:
'''Filter'''
import numpy as np

def get_center(kernel):
    '''Function that gets the center of the kernel'''
    if kernel.shape[0] % 2 == 1 and kernel.shape[1] % 2 == 1:
        return [kernel.shape[0]//2, kernel.shape[1]//2]
    return [0, 0]


def get_padding(center, kernel):
    '''Gets number of zeros to be padded in each direction'''
    if center == [0,0]:
        return ((1,1),(1,1))
    return (center[0], kernel.shape[0]-1-center[0]), (center[1], kernel.shape[1]-1-center[1])


def convolve(array, kernel):
    '''Applies convolution to a array'''
    center = get_center(kernel)
    padding = get_padding(center, kernel)
    pad_image = np.pad(array, padding)
    sha = (array.shape[0], array.shape[1], kernel.shape[0], kernel.shape[1])
    stride = np.lib.stride_tricks.as_strided(pad_image, shape=sha, strides=pad_image.strides*2)
    return np.sum(stride * kernel, axis=(2, 3))


def apply_filter(image: np.array, kernel: np.array) -> np.array:
    '''Applies filter on a 

correct code - return from a function

In [44]:
data.loc[18, "correct_code"] = True

In [45]:
show_data(data.iloc[19])

	Prefix:
'''Filter'''
import numpy as np

def get_center(kernel):
    '''Function that gets the center of the kernel'''
    if kernel.shape[0] % 2 == 1 and kernel.shape[1] % 2 == 1:
        return [kernel.shape[0]//2, kernel.shape[1]//2]
    return [0, 0]



	Fill here
	Actual code:
def get_padding(center, kernel):
	Generated code:
def get_padding(center, kernel):

Suffix:
    '''Gets number of zeros to be padded in each direction'''
    if center == [0,0]:
        return ((1,1),(1,1))
    return (center[0], kernel.shape[0]-1-center[0]), (center[1], kernel.shape[1]-1-center[1])


def convolve(array, kernel):
    '''Applies convolution to a array'''
    center = get_center(kernel)
    padding = get_padding(center, kernel)
    pad_image = np.pad(array, padding)
    sha = (array.shape[0], array.shape[1], kernel.shape[0], kernel.shape[1])
    stride = np.lib.stride_tricks.as_strided(pad_image, shape=sha, strides=pad_image.strides*2)
    return np.sum(stride * kernel, axis=(2, 3))


def app

correct code - correct function definition

In [46]:
data.loc[19, "correct_code"] = True

In [47]:
show_data(data.iloc[20])

	Prefix:
'''Filter'''
import numpy as np

def get_center(kernel):
    '''Function that gets the center of the kernel'''
    if kernel.shape[0] % 2 == 1 and kernel.shape[1] % 2 == 1:
        return [kernel.shape[0]//2, kernel.shape[1]//2]
    return [0, 0]


def get_padding(center, kernel):
    '''Gets number of zeros to be padded in each direction'''
    if center == [0,0]:
        return ((1,1),(1,1))
    return (center[0], kernel.shape[0]-1-center[0]), (center[1], kernel.shape[1]-1-center[1])

	Fill here
	Actual code:
nan
	Generated code:
nan

Suffix:

def convolve(array, kernel):
    '''Applies convolution to a array'''
    center = get_center(kernel)
    padding = get_padding(center, kernel)
    pad_image = np.pad(array, padding)
    sha = (array.shape[0], array.shape[1], kernel.shape[0], kernel.shape[1])
    stride = np.lib.stride_tricks.as_strided(pad_image, shape=sha, strides=pad_image.strides*2)
    return np.sum(stride * kernel, axis=(2, 3))


def apply_filter(image: np.array,

correct code - empty line

In [48]:
data.loc[20, "correct_code"] = True

In [49]:
show_data(data.iloc[21])

	Prefix:
'''Filter'''
import numpy as np

def get_center(kernel):
    '''Function that gets the center of the kernel'''
    if kernel.shape[0] % 2 == 1 and kernel.shape[1] % 2 == 1:
        return [kernel.shape[0]//2, kernel.shape[1]//2]
    return [0, 0]


def get_padding(center, kernel):
    '''Gets number of zeros to be padded in each direction'''
    if center == [0,0]:
        return ((1,1),(1,1))
    return (center[0], kernel.shape[0]-1-center[0]), (center[1], kernel.shape[1]-1-center[1])


def convolve(array, kernel):
    '''Applies convolution to a array'''
    center = get_center(kernel)
    padding = get_padding(center, kernel)
    pad_image = np.pad(array, padding)
    sha = (array.shape[0], array.shape[1], kernel.shape[0], kernel.shape[1])
    stride = np.lib.stride_tricks.as_strided(pad_image, shape=sha, strides=pad_image.strides*2)
    return np.sum(stride * kernel, axis=(2, 3))


def apply_filter(image: np.array, kernel: np.array) -> np.array:
    '''Applies filter on a 

wrong code - it is not correct, but the generated code does not change anything

In [50]:
data.loc[21, "correct_code"] = True

In [51]:
show_data(data.iloc[22])

	Prefix:
'''Filter'''
import numpy as np

def get_center(kernel):
    '''Function that gets the center of the kernel'''
    if kernel.shape[0] % 2 == 1 and kernel.shape[1] % 2 == 1:
        return [kernel.shape[0]//2, kernel.shape[1]//2]
    return [0, 0]


def get_padding(center, kernel):
    '''Gets number of zeros to be padded in each direction'''
    if center == [0,0]:
        return ((1,1),(1,1))
    return (center[0], kernel.shape[0]-1-center[0]), (center[1], kernel.shape[1]-1-center[1])


def convolve(array, kernel):
    '''Applies convolution to a array'''
    center = get_center(kernel)
    padding = get_padding(center, kernel)
    pad_image = np.pad(array, padding)
    sha = (array.shape[0], array.shape[1], kernel.shape[0], kernel.shape[1])
    stride = np.lib.stride_tricks.as_strided(pad_image, shape=sha, strides=pad_image.strides*2)
    return np.sum(stride * kernel, axis=(2, 3))


def apply_filter(image: np.array, kernel: np.array) -> np.array:
    '''Applies filter on a 

correct code - channel convolution

In [52]:
data.loc[22, "correct_code"] = True

In [53]:
show_data(data.iloc[23])

	Prefix:
'''Filter'''
import numpy as np

def get_center(kernel):
    '''Function that gets the center of the kernel'''
    if kernel.shape[0] % 2 == 1 and kernel.shape[1] % 2 == 1:
        return [kernel.shape[0]//2, kernel.shape[1]//2]
    return [0, 0]


def get_padding(center, kernel):
    '''Gets number of zeros to be padded in each direction'''
    if center == [0,0]:
        return ((1,1),(1,1))
    return (center[0], kernel.shape[0]-1-center[0]), (center[1], kernel.shape[1]-1-center[1])


def convolve(array, kernel):
    '''Applies convolution to a array'''
    center = get_center(kernel)
    padding = get_padding(center, kernel)
    pad_image = np.pad(array, padding)
    sha = (array.shape[0], array.shape[1], kernel.shape[0], kernel.shape[1])
    stride = np.lib.stride_tricks.as_strided(pad_image, shape=sha, strides=pad_image.strides*2)
    return np.sum(stride * kernel, axis=(2, 3))


def apply_filter(image: np.array, kernel: np.array) -> np.array:
    '''Applies filter on a 

wrong code - performing convolution twice insted of once

In [54]:
data.loc[23, "correct_code"] = False

In [55]:
show_data(data.iloc[24])

	Prefix:
from position import Coordinates
from figure import Figure
from moves import filter_moves, get_score
import math


def minimax(board, color, depth, alpha, beta, maximize):
    # with fewer figures depth should be higher todo
    best_fig = Figure()
    best_move = (Coordinates(-1, -1), False)

    # if reached the selected depth, blacks score will be returned
    if depth == 0:
        return get_score(board, "black"), (best_fig, best_move)

    if color == "white":
        opponent_color = "black"

	Fill here
	Actual code:
    else:
	Generated code:
        opponent_color = "white"

Suffix:
        opponent_color = "white"

    if maximize:
        best_score = -math.inf
    else:
        best_score = math.inf

    for x in range(8):
        for y in range(8):
            if board[x][y].get_type() != "figure" and board[x][y].get_color() == color:
                # figure has eligible moves that can be played
                moves = board[x][y].get_moves(board)
               

wrong code - missing else

In [56]:
data.loc[24, "correct_code"] = False

In [57]:
show_data(data.iloc[25])

	Prefix:
from position import Coordinates
from figure import Figure

	Fill here
	Actual code:
from moves import filter_moves, get_score
	Generated code:
from filter_moves <|endoftext|>

Suffix:
import math


def minimax(board, color, depth, alpha, beta, maximize):
    # with fewer figures depth should be higher todo
    best_fig = Figure()
    best_move = (Coordinates(-1, -1), False)

    # if reached the selected depth, blacks score will be returned
    if depth == 0:
        return get_score(board, "black"), (best_fig, best_move)

    if color == "white":
        opponent_color = "black"
    else:
        opponent_color = "white"

    if maximize:
        best_score = -math.inf
    else:
        best_score = math.inf

    for x in range(8):
        for y in range(8):
            if board[x][y].get_type() != "figure" and board[x][y].get_color() == color:
                # figure has eligible moves that can be played
                moves = board[x][y].get_moves(board)
                

wrong code - incorrect import

In [58]:
data.loc[25, "correct_code"] = False

In [59]:
show_data(data.iloc[26])

	Prefix:
from position import Coordinates
from figure import Figure
from moves import filter_moves, get_score
import math


def minimax(board, color, depth, alpha, beta, maximize):
    # with fewer figures depth should be higher todo

	Fill here
	Actual code:
    best_fig = Figure()
	Generated code:
    if depth == 0:

Suffix:
    best_move = (Coordinates(-1, -1), False)

    # if reached the selected depth, blacks score will be returned
    if depth == 0:
        return get_score(board, "black"), (best_fig, best_move)

    if color == "white":
        opponent_color = "black"
    else:
        opponent_color = "white"

    if maximize:
        best_score = -math.inf
    else:
        best_score = math.inf

    for x in range(8):
        for y in range(8):
            if board[x][y].get_type() != "figure" and board[x][y].get_color() == color:
                # figure has eligible moves that can be played
                moves = board[x][y].get_moves(board)
                figure = boar

wrong code

In [60]:
data.loc[26, "correct_code"] = False

In [61]:
show_data(data.iloc[27])

	Prefix:
from position import Coordinates
from figure import Figure
from moves import filter_moves, get_score
import math


def minimax(board, color, depth, alpha, beta, maximize):
    # with fewer figures depth should be higher todo
    best_fig = Figure()
    best_move = (Coordinates(-1, -1), False)

    # if reached the selected depth, blacks score will be returned
    if depth == 0:
        return get_score(board, "black"), (best_fig, best_move)

    if color == "white":
        opponent_color = "black"
    else:
        opponent_color = "white"

    if maximize:
        best_score = -math.inf
    else:
        best_score = math.inf

    for x in range(8):
        for y in range(8):
            if board[x][y].get_type() != "figure" and board[x][y].get_color() == color:
                # figure has eligible moves that can be played
                moves = board[x][y].get_moves(board)
                figure = board[x][y]
                moves = filter_moves(moves, board, figure)

   

wrong code - incorrect if statement

In [62]:
data.loc[27, "correct_code"] = False

In [63]:
show_data(data.iloc[28])

	Prefix:
from position import Coordinates
from figure import Figure
from moves import filter_moves, get_score
import math


def minimax(board, color, depth, alpha, beta, maximize):
    # with fewer figures depth should be higher todo
    best_fig = Figure()
    best_move = (Coordinates(-1, -1), False)


	Fill here
	Actual code:
    # if reached the selected depth, blacks score will be returned
	Generated code:
    for x in range(8):

Suffix:
    if depth == 0:
        return get_score(board, "black"), (best_fig, best_move)

    if color == "white":
        opponent_color = "black"
    else:
        opponent_color = "white"

    if maximize:
        best_score = -math.inf
    else:
        best_score = math.inf

    for x in range(8):
        for y in range(8):
            if board[x][y].get_type() != "figure" and board[x][y].get_color() == color:
                # figure has eligible moves that can be played
                moves = board[x][y].get_moves(board)
                figure = 

wrong code - no line or comment should have been generated

In [64]:
data.loc[28, "correct_code"] = False

In [65]:
show_data(data.iloc[29])

	Prefix:
from position import Coordinates
from figure import Figure
from moves import filter_moves, get_score
import math


def minimax(board, color, depth, alpha, beta, maximize):
    # with fewer figures depth should be higher todo
    best_fig = Figure()
    best_move = (Coordinates(-1, -1), False)

    # if reached the selected depth, blacks score will be returned
    if depth == 0:
        return get_score(board, "black"), (best_fig, best_move)

    if color == "white":
        opponent_color = "black"
    else:

	Fill here
	Actual code:
        opponent_color = "white"
	Generated code:
        opponent_color = color

Suffix:

    if maximize:
        best_score = -math.inf
    else:
        best_score = math.inf

    for x in range(8):
        for y in range(8):
            if board[x][y].get_type() != "figure" and board[x][y].get_color() == color:
                # figure has eligible moves that can be played
                moves = board[x][y].get_moves(board)
                f

wrong code - incorrect color assignment

In [66]:
data.loc[29, "correct_code"] = False

In [67]:
show_data(data.iloc[30])

	Prefix:
from position import Coordinates
from figure import Figure
from moves import filter_moves, get_score
import math


def minimax(board, color, depth, alpha, beta, maximize):
    # with fewer figures depth should be higher todo
    best_fig = Figure()
    best_move = (Coordinates(-1, -1), False)

    # if reached the selected depth, blacks score will be returned
    if depth == 0:
        return get_score(board, "black"), (best_fig, best_move)

    if color == "white":
        opponent_color = "black"
    else:
        opponent_color = "white"

    if maximize:
        best_score = -math.inf
    else:
        best_score = math.inf

    for x in range(8):
        for y in range(8):
            if board[x][y].get_type() != "figure" and board[x][y].get_color() == color:
                # figure has eligible moves that can be played
                moves = board[x][y].get_moves(board)
                figure = board[x][y]
                moves = filter_moves(moves, board, figure)

   

wrong code - no line

In [68]:
data.loc[30, "correct_code"] = False

In [69]:
show_data(data.iloc[31])

	Prefix:
from position import Coordinates
from figure import Figure
from moves import filter_moves, get_score
import math


def minimax(board, color, depth, alpha, beta, maximize):
    # with fewer figures depth should be higher todo
    best_fig = Figure()
    best_move = (Coordinates(-1, -1), False)

    # if reached the selected depth, blacks score will be returned
    if depth == 0:
        return get_score(board, "black"), (best_fig, best_move)

    if color == "white":
        opponent_color = "black"
    else:
        opponent_color = "white"

    if maximize:
        best_score = -math.inf
    else:
        best_score = math.inf

    for x in range(8):
        for y in range(8):
            if board[x][y].get_type() != "figure" and board[x][y].get_color() == color:
                # figure has eligible moves that can be played
                moves = board[x][y].get_moves(board)
                figure = board[x][y]
                moves = filter_moves(moves, board, figure)

   

wrong code - empty line

In [70]:
data.loc[31, "correct_code"] = False

### Summary

In [4]:
print_results(data)

hand_label = data['correct_code'].value_counts()
print(f"Hand label:\n{hand_label.to_string(index=True, header=False)}")
print(f"{(hand_label[True]/(hand_label[True] + hand_label[False]))*100} %")

data.to_csv("data.csv", index=False)

CHRF:
26.57 %

Correct syntax:
True     16
False    11
nan       5
60.71 %

Exact match:
False    22
nan       5
True      5
21.43 %

rougeL:
27.57 %

Hand label:
False    20
True     12
37.5 %


### Result

The results from testing the Tiny Starcoder model for code completion on a dataset containing empty lines intended for completion indicate that its performance is limited. Both CHRF and ROUGE scores reflect an accuracy of around 27%, suggesting room for improvement. On a positive note, the model achieves a syntax accuracy of over 60%, which shows that it can generally maintain correct syntax. However, only 37% of examples are fully completed as intended. 

For higher performance, models like the full-size Starcoder or CodeLlama would be preferable, though they are too computationally demanding to run effectively on my current setup.